### ISSUE TITLE
NullPointerException when rescheduling notifications

### ISSUE THREAD

**Issue by liamheneghan** [invertase/react-native-firebase#1013]
### Issue
I have received some crash reports which originate from the notifications module of the React Native Firebase library.

After looking at the source it seems`rescheduleNotifications()` (called from the boot receiver) always calls `scheduleNotification(...)` with a `null` Promise causing the exception posted below.

To replicate:
1) Schedule a notification for some future date
2) Reboot the Android device

```
Fatal Exception: java.lang.RuntimeException: Unable to start receiver io.invertase.firebase.notifications.RNFirebaseNotificationsRebootReceiver: java.lang.NullPointerException: Attempt to invoke interface method 'void com.facebook.react.bridge.Promise.resolve(java.lang.Object)' on a null object reference
       at android.app.ActivityThread.handleReceiver(ActivityThread.java:3194)
       at android.app.ActivityThread.-wrap17(Unknown Source)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1672)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loop(Looper.java:164)
       at android.app.ActivityThread.main(ActivityThread.java:6501)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)

Caused by java.lang.NullPointerException: Attempt to invoke interface method 'void com.facebook.react.bridge.Promise.resolve(java.lang.Object)' on a null object reference
       at io.invertase.firebase.notifications.RNFirebaseNotificationManager.scheduleNotification(RNFirebaseNotificationManager.java:708)
       at io.invertase.firebase.notifications.RNFirebaseNotificationManager.rescheduleNotifications(RNFirebaseNotificationManager.java:187)
       at io.invertase.firebase.notifications.RNFirebaseNotificationsRebootReceiver.onReceive(RNFirebaseNotificationsRebootReceiver.java:16)
       at android.app.ActivityThread.handleReceiver(ActivityThread.java:3187)
       at android.app.ActivityThread.-wrap17(Unknown Source)
       at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1672)
       at android.os.Handler.dispatchMessage(Handler.java:106)
       at android.os.Looper.loop(Looper.java:164)
       at android.app.ActivityThread.main(ActivityThread.java:6501)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
```

### Environment

Android

1. Application Target Platform:

Android

2. Development Operating System:

-

3. Build Tools:

-

4. React Native version:

0.53.3

5. RNFirebase Version:

4.0.3

6. Firebase Module:

Notifications

**Comment by chrisbianca** [comment]
Thanks for flagging - I've just pushed up a fix which will be released as part of v4.0.6 later this week

**Comment by matttti** [comment]
I also had that issue, and played around in the code last week (version 4.0.4). I discovered a follow-up bug: all scheduled notifications are fired instantly after rebooting the device, no matter of their schedule date.

`Double fireDate = schedule.getDouble("fireDate");` is always `0` after reboot.

The reason is that number values created by `BundleJSONConverter` are parsed as `Long` (i guess in case they have no signs of being floating numbers)

For now, I helped myself with this patch:

```
     Double fireDate = schedule.getDouble("fireDate");
+    if (fireDate == 0) {
+      Long fireDateLong = schedule.getLong("fireDate");
+      fireDate = fireDateLong.doubleValue();
+    }
```

I cannot update to 4.0.6 at the moment. So in case this hasn't been addressed already, i hope is of help to you.
